src/cdk-experimental/menu/menu.spec.ts

```diff
@@ -68,22 +68,126 @@ describe('Menu', () => {
+      TestBed.configureTestingModule({
+        imports: [CdkMenuModule],
+        declarations: [MenuWithNestedGroup],
+      }).compileComponents();
+
+      fixture = TestBed.createComponent(MenuWithNestedGroup);
+      fixture.detectChanges();
+
+      menu = fixture.debugElement.query(By.directive(CdkMenu)).injector.get(CdkMenu);
+
+      menuItems = fixture.debugElement
+        .queryAll(By.directive(CdkMenuItemCheckbox))
+        .map(element => element.injector.get(CdkMenuItemCheckbox));
+    }));
+
+    it('should not emit change from root menu ', () => {
+      const spy = jasmine.createSpy('changeSpy for root menu');
+      menu.change.subscribe(spy);
+
+      menuItems.forEach(menuItem => menuItem.trigger());
+
+      expect(spy).toHaveBeenCalledTimes(0);
+    });
+  });
+
+  describe('with inner group render delayed', () => {
+    let fixture: ComponentFixture<MenuWithConditionalGroup>;
+    let menuItems: CdkMenuItemCheckbox[];
+    let menu: CdkMenu;
+
+    const getMenuItems = () => {
+      return fixture.debugElement
+        .queryAll(By.directive(CdkMenuItemCheckbox))
+        .map(element => element.injector.get(CdkMenuItemCheckbox));
+    };
+
+    beforeEach(async(() => {
+      TestBed.configureTestingModule({
+        imports: [CdkMenuModule],
+        declarations: [MenuWithConditionalGroup],
+      }).compileComponents();
+
+      fixture = TestBed.createComponent(MenuWithConditionalGroup);
+
+      fixture.detectChanges();
+
+      menu = fixture.debugElement.query(By.directive(CdkMenu)).injector.get(CdkMenu);
+
+      menuItems = getMenuItems();
+    }));
+
+    it('should not emit after the group mounts', () => {
+      const spy = jasmine.createSpy('cdkMenu change');
+      menu.change.subscribe(spy);
+
+      menuItems[0].trigger();
+      fixture.componentInstance.show = true;
+      fixture.detectChanges();
+
+      menuItems = getMenuItems();
+      menuItems[0].trigger();
+      fixture.detectChanges();
+
+      expect(spy).withContext('Expected initial trigger only').toHaveBeenCalledTimes(1);
+    });
+  });
 });
 
 @Component({
   template: `
     <ul cdkMenu>
       <li role="none">
-        <button checked="true" role="menuitemcheckbox" cdkMenuItem>
+        <button checked="true" cdkMenuItemCheckbox>
           first
         </button>
       </li>
       <li role="none">
-        <button role="menuitemcheckbox" cdkMenuItem>
+        <button cdkMenuItemCheckbox>
           second
         </button>
       </li>
     </ul>
   `,
 })
 class MenuCheckboxGroup {}
+
+@Component({
+  template: `
+    <ul cdkMenu>
+      <li>
+        <ul cdkMenuGroup>
+          <li><button cdkMenuCheckbox>first</button></li>
+        </ul>
+      </li>
+    </ul>
+  `,
+})
+class MenuWithNestedGroup {}
+
+@Component({
+  template: `
+    <ul cdkMenu>
+      <ng-template [ngIf]="!show">
+        <li><button cdkMenuItemCheckbox>first</button></li>
+      </ng-template>
+      <ng-template [ngIf]="show">
```

Comment:
Ah, now I see -- 'show = true' is what mounts the inner group. However, does the test depend on 'first' being hidden when show is true? If not, I'd remove the condition from 'first'.

I'd use a clearer name for 'show' so the test code is clearer -- maybe 'showInnerGroup' or 'renderInnerGroup'?

Also, is there a reason you're using the more verbose [ngIf] syntax? (If there is, I'd add a comment explaining the reason, unless it's part of CDK convention.)